server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+// router.post('/present/:user/full', hasRole('mentor'),controller.verifyFullAttendance);
+exports.verifyFullAttendance = function(req,res){
+    var userId = req.params.id;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        return getPresent(user._id, isoDateToTime(new Date()), classYear._id, function(userAttendance){
+            if (!userAttendance.length){
+                return res.send(400, 'No attendance found');
+            }
+            for (var a = 0; a < userAttendance.length; a++){
+                if (!userAttendance.smallgroup){
+                    userAttendance[a].verified = true;
+                    userAttendance[a].save();
+                }
+            }
+            return res.send(200);
+
+        });
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Verify users small group attendance given submission Id, available only to mentors
+// TODO: verify that mentor is part of the smallgroup
+// router.post('/present/:user/small', auth.hasRole('mentor'), controller.verifySmallAttendance);
+exports.verifySmallAttendance = function(req,res){
+    var userId = req.params.id;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        return getPresent(user._id, isoDateToTime(new Date()), classYear._id, function(userAttendance){
+            if (!userAttendance.length){
+                return res.send(400, 'No attendance found');
+            }
+            for (var a = 0; a < userAttendance.length; a++){
+                if (userAttendance.smallgroup){
+                    userAttendance[a].verified = true;
+                    userAttendance[a].save();
+                }
+            }
+            return res.send(200);
+        });
+    });
+};
+// *******************************************************
+
+
+// *******************************************************
+// Get all attendance for a specific user (or current user) in the current classyear
+// router.get('/present/me', auth.isAuthenticated(), controller.getAttendanceMe);
+// router.get('/present/:user', auth.hasRole('mentor'), controller.getAttendance);
+
+// Get all attendance for a specific user (or current user) in the current class year
+var getAttendance = function(userId, classYearId, cb){
+    var callback = cb || function(){};
+    Attendance.find({user:userId, classYear:classYearId}, function (err, attendance) {
+        return callback(attendance);
+    });
+};
+exports.getAttendance = function(req, res) {
+    var userId = req.params.user;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        getAttendance(userId, classYearId, function(userAttendance){
+            return res.json(userAttendance);
+        });
+    });
+};
+exports.getAttendanceMe = function(req, res) {
```

Comment:
This is simply a wrapper to enable router.get('/present/me'). It is effectively the same as using the logged in user, but enables /present/me to have weaker permissions than /present/:user (since users can only get their own if not a mentor)